### PR TITLE
Improve error message in file comparison-type tests

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Previously if the subfile wasn't in the HDF5 file, maybe because of a typo, it
was not clear _why_ the test failed. Now we report the subfile we are trying to
find, that the check failed because the subfile wasn't there, and what the
available subfiles are in all the HDF5 files that matched the glob.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tools/CheckOutputFiles.py [hunk 1]
```diff
@@ -136,17 +142,27 @@ def perform_check(self, run_directory):
...
         for filename in glob.glob(os.path.join(run_directory, self.h5_glob)):
             logging.info("Checking file: " + filename)
             with self.unit_test.subTest(filename=filename):
                 with h5py.File(filename, 'r') as check_h5:
+                    files_and_entities = (files_and_entities + filename +
+                                          ": " + str(list(check_h5.keys())) +
+                                          "\n")
+                    found_test_entity = found_test_entity or (
+                        self.test_h5_entity in check_h5)
+                    found_expected_entity = found_test_entity or (
```
Comment (nilsvu) [inline tools/CheckOutputFiles.py]: `found_test_entity` -> `found_expected_entity`

File: tools/CheckOutputFiles.py [hunk 2]
```diff
@@ -155,8 +171,16 @@ def perform_check(self, run_directory):
...
-            checks_performed, "No checks performed for glob: " +
-            os.path.join(run_directory, self.h5_glob))
+            found_test_entity,
+            "Failed to find the subfile '" + self.test_h5_entity +
+            "' using glob:\n" + os.path.join(run_directory, self.h5_glob) +
+            "\nFiles and entities:\n" + files_and_entities)
+        self.unit_test.assertTrue(
+            (self.expected_h5_entity is None) or found_expected_entity,
+            "Failed to find the expected entity/subfile (i.e. the data set to "
+            "which the test data is compared to) '" + self.test_h5_entity +
```
Comment (nilsvu) [inline tools/CheckOutputFiles.py]: `self.expected_h5_entity`

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Squashed (since two small typos) and rebase. Thanks for the review!

File: tools/CheckOutputFiles.py [hunk 3]
```diff
@@ -155,8 +171,16 @@ def perform_check(self, run_directory):
                         self.check_h5_file(check_h5, self.test_h5_entity,
                                            self.expected_h5_entity)
         self.unit_test.assertTrue(
-            checks_performed, "No checks performed for glob: " +
-            os.path.join(run_directory, self.h5_glob))
+            found_test_entity,
+            "Failed to find the subfile '" + self.test_h5_entity +
+            "' using glob:\n" + os.path.join(run_directory, self.h5_glob) +
+            "\nFiles and entities:\n" + files_and_entities)
+        self.unit_test.assertTrue(
```
Comment (nikwit) [inline tools/CheckOutputFiles.py]: It looks like python tries to create the error message regardless of whether the assert fails or not which is why the output checks fail.
```
if self.expected_h5_entity is not None:
        self.unit_test.assertTrue(found_expected_entity, ...)
```
fixed it for me. (It is probably more readable anyway)

**Review by nikwit — CHANGES_REQUESTED** [review]
Looks good, one fix.

File: tools/CheckOutputFiles.py (hunk 3 above)
Comment (nilsdeppe) [reply tools/CheckOutputFiles.py]: ah yep :) And agreed, that is easier to read! :) Squashed and rebased

**Review by nilsvu — APPROVED** [review]

**Pull request merged**